src/components/interaction-designer/Connection.vue

```diff
@@ -4,115 +4,119 @@
     :reposition-hook="repositionHook" />
 </template>
 
-<script lang="js">
+<script lang="ts">
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types,@typescript-eslint/strict-boolean-expressions */
+import {mixins} from 'vue-class-component'
+import {Component, Prop} from 'vue-property-decorator'
+import {namespace} from 'vuex-class'
 import LeaderLine from 'leader-line'
 import {set} from 'lodash'
-import {mapGetters, mapMutations, mapState} from 'vuex'
+import {IBlock, IBlockExit} from '@floip/flow-runner'
+import Lang from '@/lib/filters/lang'
 
 const lightColor = '#6897BB'
 const darkColor = '#30516a'
 const disconnectionColor = '#dc3545'
 
-export const Connection = {
-  props: {
-    exit: Object,
+const builderNamespace = namespace('builder')
 
-    repaintCacheKeyGenerator: Function,
-    source: Object,
-    target: Object,
-    position: Object,
-  },
+@Component({})
+export class Connection extends mixins(Lang) {
+  @Prop({type: Object, required: true}) readonly exit!: IBlockExit
+  @Prop({type: Function, required: true}) readonly repaintCacheKeyGenerator!: Function
+  @Prop({type: Object, required: true}) readonly source!: IBlock | null
+  @Prop({type: Object, required: true}) readonly target!: IBlock | null
+  @Prop({type: Object, required: true}) readonly position!: { x: number, y: number } | null
 
-  data() {
+  isPermanentlyActive = false
+  line: any = null
```

Comment:
I restored the original comment but set `undefined` [to make it non-reactive](https://class-component.vuejs.org/guide/class-component.html#data); otherwise, TS is unable to find the property.